library/src/main/java/com/android/mms/transaction/PushReceiver.java

```diff
@@ -192,7 +192,8 @@ protected Void doInBackground(Intent... intents) {
                                 }
 
                                 if (useSystem) {
-                                    DownloadManager.getInstance().downloadMultimediaMessage(mContext, location, uri, true);
+                                    int subId = intent.getIntExtra("subscription", Settings.DEFAULT_SUBSCRIPTION_ID);
```

Comment:
Is there a constant for this, or any documentation around this `subscription` extra?

Reply:
Couldn't find a constant. There is documentation in android Telephony.java
*   <li><em>"subscription"</em> - An optional long value of the subscription id which
*   received the message.</li>